src/spetlr/configurator/configurator.py

```diff
@@ -353,6 +353,13 @@ def register(self, key: str, value: TcValue) -> str:
         ):
             # if both are dicts, update the contents.
             self._raw_resource_details[key].update(value)
+
+            # This is how to clear a key if the keyvalue is None
+            self._raw_resource_details[key] = {
+                k: v
+                for k, v in self._raw_resource_details[key].items()
+                if v is not None
+            }
         else:
```

Comment:
maybe add:
if not self._raw_resource_details[key]:
self._raw_resource_details.pop(key)

If the dictionary ended up empty by this filtering operation, remove it entirely.